commands/orchestrationcommands/stackcommands/adopt_test.go

```diff
@@ -36,3 +40,57 @@ func TestAdoptServiceClientType(t *testing.T) {
 	actual := cmd.ServiceClientType()
 	th.AssertEquals(t, expected, actual)
 }
+
+func TestAdoptPreCSV(t *testing.T) {
+	cmd := &commandAbandon{
+		Ctx: &handler.Context{
+			ServiceClient: client.ServiceClient(),
+		},
+	}
+
+	resource := &handler.Resource{
+		Params: &paramsAdopt{},
+	}
+
+	expected := "{\"ID\":\"16ef0584-4458-41eb-87c8-0dc8d5f66c87\",\"Links\":[{\"Href\":\"http://168.28.170.117:8004/v1/98606384f58drad0bhdb7d02779549ac/stacks/stackcreated/16ef0584-4458-41eb-87c8-0dc8d5f66c87\",\"Rel\":\"self\"}]}"
```

Comment:
The PR looks great. There's just one thing confusing me: I see above you're flattening the `Links` field in the `adopt` `PreCSV`, and below you're calling that method, so why are there arrays in the JSONified `PreCSV` output above?